### PR TITLE
[FIX] account: section_and_note fields not working

### PR DESCRIPTION
**Pull request by h4818**
This seems to be caused by changes in odoo/odoo#97984 and odoo/odoo#98234 The tests didn't seem to be executing since they had legacy dependencies

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js [hunk 1]
```diff
@@ -42,32 +43,40 @@ export class SectionAndNoteListRenderer extends ListRenderer {
...
-SectionAndNoteListRenderer.template = "account.sectionAndNoteListRenderer";
-SectionAndNoteListRenderer.recordRowTemplate = "account.SectionAndNoteListRenderer.RecordRow";
 
-export class SectionAndNoteFieldOne2Many extends X2ManyField {
-    setup() {
-        super.setup();
-        if (this.viewMode === 'list') {
-            this.Renderer = SectionAndNoteListRenderer;
+    getSectionColumns(columns) {
+        const sectionColumns = columns.filter((col) => col.widget === "handle");
```
Comment (william-andre) [inline addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js:61]: What? A column is part of the section if an only if it is a handle?

File: addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js [hunk 2]
```diff
@@ -42,32 +43,40 @@ export class SectionAndNoteListRenderer extends ListRenderer {
...
+        let colspan = columns.length - sectionColumns.length;
+        if (this.props.activeActions.onDelete) {
+            colspan++;
```
Comment (william-andre) [inline addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js:64]: Aren't you supposed to remove a span? 
Don't we want to have the possibility to remove the line?

**Review by william-andre — CHANGES_REQUESTED** [review]

File: addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js (hunk 1 above)
Comment (h4818) [reply addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js:61]: Yes, sequence and name should be the only columns

File: addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js (hunk 2 above)
Comment (h4818) [reply addons/account/static/src/components/section_and_note_fields_backend/section_and_note_fields_backend.js]: With a delete button / or without, the colspan should be the same

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**